opm/autodiff/AquiferCarterTracy.hpp

```diff
@@ -0,0 +1,362 @@
+                   pa0_     , // initial aquifer pressure
+                   gravity_ ; // gravitational acceleration
+
+            Eval W_flux_;
+
+
+            inline void get_influence_table_values(Scalar& pitd, Scalar& pitd_prime, const Scalar& td)
+            {
+                // We use the opm-common numeric linear interpolator
+                pitd = Opm::linearInterpolation(aquct_data_.td, aquct_data_.pi, td);
+                pitd_prime = Opm::linearInterpolationDerivative(aquct_data_.td, aquct_data_.pi, td);
+            }
+
+            inline void init_quantities(const Aquancon::AquanconOutput& connection)
+            {
+                // We reset the cumulative flux at the start of any simulation, so, W_flux = 0
+                W_flux_ = 0.;
+
+                // We next get our connections to the aquifer and initialize these quantities using the initialize_connections function
+                initialize_connections(connection);
+
+                calculate_aquifer_condition();
+
+                calculate_aquifer_constants();
+
+                pressure_previous_.resize(cell_idx_.size(), 0.);
+                pressure_current_.resize(cell_idx_.size(), 0.);
+                Qai_.resize(cell_idx_.size(), 0.0);
+            }
+
+            inline void update_cell_pressure(std::vector<Eval>& pressure_water, const int idx, const IntensiveQuantities& intQuants)
+            {
+                const auto& fs = intQuants.fluidState();
+                pressure_water.at(idx) = fs.pressure(waterPhaseIdx);
+            }
+
+            inline void update_cell_density(const int idx, const IntensiveQuantities& intQuants)
+            {
+                const auto& fs = intQuants.fluidState();
+                rhow_.at(idx) = fs.density(waterPhaseIdx);
+            }
+
+            inline Scalar dpai(int idx)
+            {
+                Scalar dp = pa0_ + rhow_.at(idx).value()*gravity_*(cell_depth_.at(idx) - aquct_data_.d0) - pressure_previous_.at(idx).value();
+                return dp;
+            }
+
+            // This function implements Eqs 5.8 and 5.9 of the EclipseTechnicalDescription
+            inline void calculate_a_b_constants(Scalar& a, Scalar& b, const int idx, const SimulatorTimerInterface& timer)
+            {
+                Scalar td_plus_dt = (timer.currentStepLength() + timer.simulationTimeElapsed()) / Tc_;
+                Scalar td = timer.simulationTimeElapsed() / Tc_;
+                Scalar PItdprime = 0.;
+                Scalar PItd = 0.;
+                get_influence_table_values(PItd, PItdprime, td_plus_dt);
+                a = 1.0/Tc_ * ( (beta_ * dpai(idx)) - (W_flux_.value() * PItdprime) ) / ( PItd - td*PItdprime );
+                b = beta_ / (Tc_ * ( PItd - td*PItdprime));
+            }
+
+            // This function implements Eq 5.7 of the EclipseTechnicalDescription       
+            inline void calculate_inflow_rate(int idx, const SimulatorTimerInterface& timer)
```

Comment:
If you would like to, you can use camelCase for the functions. It is more consistent with the rest part of OPM.